File: src/interpreter/variable.cpp
#include "variable.h"

VariableType VariableType::Auto = VariableType("Auto", "auto");
VariableType VariableType::Int = VariableType("Int", "int");
VariableType VariableType::String = VariableType("String", "str");
VariableType VariableType::Any = VariableType("Any", "any");
VariableType VariableType::Function = VariableType("Function", "func");
VariableType VariableType::Type = VariableType("Type", "type");

VariableType VariableType::Array = VariableType("Array", 1);
VariableType VariableType::Dict = VariableType("Dict", 2);
VariableType VariableType::Object = VariableType("Object", 3);

std::map<std::string, VariableType> VariableType::s_Values = {
    { VariableType::Auto.name, VariableType::Auto },
    { VariableType::Int.name, VariableType::Int },
    { VariableType::String.name, VariableType::String },
    { VariableType::Any.name, VariableType::Any },
    { VariableType::Function.name, VariableType::Function },
    { VariableType::Type.name, VariableType::Type },

    { VariableType::Array.name, VariableType::Array },
    { VariableType::Dict.name, VariableType::Dict },
    { VariableType::Object.name, VariableType::Object }
};

File: oldsrc/interpreter/variable.rs
pub mod language {
    pub struct VariableType {}
    impl VariableType {
        pub const AUTO: &str =     "auto";
        pub const INT: &str =      "int";
        pub const STRING: &str =   "str";
        pub const ANY: &str =      "any";
        pub const FUNCTION: &str = "func";
        pub const TYPE: &str =     "type";
        
        pub const ARRAY: i32 =     1;
        pub const DICT: i32 =      2;
        pub const OBJECT: i32 =    3; // Class, data structure, etc.
    }
}
